src/cascadia/TerminalSettingsEditor/Actions.cpp

```diff
@@ -4,61 +4,261 @@
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Data;
 using namespace winrt::Windows::UI::Xaml::Navigation;
 using namespace winrt::Microsoft::Terminal::Settings::Model;
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
+    KeyBindingViewModel::KeyBindingViewModel(const Control::KeyChord& keys, const Model::Command& cmd) :
+        _Keys{ keys },
+        _KeyChordText{ Model::KeyChordSerialization::ToString(keys) },
+        _Command{ cmd }
+    {
+        // Add a property changed handler to our own property changed event.
+        // This propagates changes from the settings model to anybody listening to our
+        //  unique view model members.
+        PropertyChanged([this](auto&&, const PropertyChangedEventArgs& args) {
+            const auto viewModelProperty{ args.PropertyName() };
+            if (viewModelProperty == L"Keys")
+            {
+                _KeyChordText = Model::KeyChordSerialization::ToString(_Keys);
+                _NotifyChanges(L"KeyChordText");
+            }
+            else if (viewModelProperty == L"IsContainerFocused" ||
+                     viewModelProperty == L"IsEditButtonFocused" ||
+                     viewModelProperty == L"IsHovered" ||
+                     viewModelProperty == L"IsAutomationPeerAttached" ||
+                     viewModelProperty == L"IsInEditMode")
+            {
+                _NotifyChanges(L"ShowEditButton");
+            }
+        });
+    }
+
+    bool KeyBindingViewModel::ShowEditButton() const noexcept
+    {
+        return (IsContainerFocused() || IsEditButtonFocused() || IsHovered() || IsAutomationPeerAttached()) && !IsInEditMode();
+    }
+
+    void KeyBindingViewModel::ToggleEditMode()
+    {
+        // toggle edit mode
+        IsInEditMode(!_IsInEditMode);
+        if (_IsInEditMode)
+        {
+            // if we're in edit mode,
+            // pre-populate the text box with the current keys
+            _ProposedKeys = KeyChordText();
+            _NotifyChanges(L"ProposedKeys");
+        }
+    }
+
+    void KeyBindingViewModel::AttemptAcceptChanges()
+    {
+        auto args{ make_self<RebindKeysEventArgs>(_Keys, _Keys) };
+        try
+        {
+            // Attempt to convert the provided key chord text
+            const auto newKeyChord{ KeyChordSerialization::FromString(_ProposedKeys) };
+            args->NewKeys(newKeyChord);
+        }
+        catch (hresult_invalid_argument)
+        {
+            // Converting the text into a key chord failed
+            // TODO CARLOS:
+            //  This is tricky. I still haven't found a way to reference the
+            //  key chord text box. It's hidden behind the data template.
+            //  Ideally, some kind of notification would alert the user, but
+            //  to make it look nice, we need it to somehow target the text box.
+        }
+        _RebindKeysRequestedHandlers(*this, *args);

```

Comment:
Good catch

<details>

<summary>Full explanation</summary>

So here's all the cases:
1. successful parse
   - same key chord as before
      - `_ViewModelRebindKeysHandler`: **toggle edit mode**
   - different key chord as before
      - `_ViewModelRebindKeysHandler`
         - a key binding is already using that key chord
            - notify the user; If the user confirms, 
               - update settings model
               - update view model
               - **toggle edit mode**
         - otherwise
            - update settings model
            - update view model
            - **toggle edit mode**
2. invalid key chord
   - (TODO) notify the user
   - **DO NOT toggle edit mode**

</details>